Library/Homebrew/utils/shell_env.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set +o posix
+
+KeepKeys=(
+  BASH
+  BASH_ARGC
+  BASH_ARGV
+  BREW_FILE_DIRECTORY
+  USR_LOCAL_BREW_FILE_DIRECTORY
+  USR_LOCAL_HOMEBREW_REPOSITORY
+  HOMEBREW_*
+  HOME
+  PWD
+  LANG
+  PATH
+  TERM
+)
+
+prune-shell-env() {
+  file=/tmp/homebrew-env
+
+  IFS="="
+  for i in ${KeepKeys[*]}
+  do
+  
+    if [[ -n "$verbose" ]]
+    then
+      echo "Keeping: $i"
+    fi
+    
+    while read -r key value
+    do
+    
+      # 'shellcheck' advised to 'Quote the rhs of == in [[ ]] to prevent glob matching.' but we want globs.
+      if [[ $key == $i ]]
+      then
+      
+        if [[ -n "$verbose" ]]
+        then
+          echo "$key=$value"
+        fi
+
+        export "$key=$value"
+      fi
+    done < "$file"
+
+  done
+
+  # Need to reset IFS to default value for later shell functions like 'printf -v HOMEBREW_MACOS_VERSION_NUMERIC ...' to work
+  unset IFS
+
+}
+
+verbose=''
+for option in "$@"
+  do
+    case "$option" in
+      --verbose) verbose=1 ;;
+    esac
+done
```

Comment:
I reckon you can kill the `--verbose` logic. If you want to leave some sort of debugging logic in make it conditional on an environment variable i.e. `HOMEBREW_ENV_DEBUG`